authors/apps/profiles/views.py

```diff
@@ -15,32 +18,54 @@ class ProfileRetrieveUpdateView(generics.GenericAPIView):
     """
     Class for retriving  and updating user profile
     """
-    permission_classes = (permissions.IsAuthenticated,IsOwnerOrReadOnly)
+
+    permission_classes = (permissions.IsAuthenticated, IsOwnerOrReadOnly)
     renderer_classes = (ProfileRenderer,)
     serializer_class = ProfileSerializer
 
     def get(self, request, *args, **kwargs):
-        #Function to retrieve user profile
+        # Function to retrieve user profile
         serializer = self.serializer_class(
-            get_object_or_404(Profile, user=self.kwargs.get("username")))
+            get_object_or_404(Profile, user=self.kwargs.get("username"))
+        )
         return Response(serializer.data, status=status.HTTP_200_OK)
 
     def put(self, request, *args, **kwargs):
         # Function to update user profile
 
         try:
-            user = get_user_model().objects.get(username=self.kwargs.get("username"))
+            user = get_user_model().objects.get(
+                username=self.kwargs.get("username")
+            )
         except ObjectDoesNotExist:
             raise Http404()
 
         profile = Profile.objects.get(user=user.username)
 
         if not profile.user.pk == request.user.id:
-            return Response({"detail": "You don't have permissions to update this user"}, 
-            status=status.HTTP_403_FORBIDDEN)
+            return Response(
+                {"detail": "You don't have permissions to update this user"},
+                status=status.HTTP_403_FORBIDDEN,
+            )
         serializer = self.serializer_class(
-            profile, data=request.data["profile"], partial=True)
+            profile, data=request.data["profile"], partial=True
+        )
         if serializer.is_valid(raise_exception=True):
             serializer.save()
             return Response(serializer.data, status=status.HTTP_200_OK)
 
+
+class ListProfiles(APIView):
+    """
+    Class to list author profiles
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+    def get(self, request, *args, **kwargs):
+
+        articles = Article.objects.distinct("author").all()
+        authors = Profile.objects.filter(
```

Comment:
well done, @bisonlou, this is some nice implementation. I have actually tested the PR and Only users who have posted at least one article are regarded as authors.

Reply:
And furthermore, it would be a great idea add an `articles_count` in the author profiles returned. This would show the number of articles published by each user.